// nuxi/dist/chunks/build.mjs
import { r as resolve, a as relative } from '../shared/nuxi-edge.c25bd819.mjs';
import { c as consola } from '../shared/nuxi-edge.1a52f0e8.mjs';
import { w as writeTypes } from '../shared/nuxi-edge.02b7bbbc.mjs';
import { l as loadKit } from '../shared/nuxi-edge.d8a764d5.mjs';
import { c as clearDir } from '../shared/nuxi-edge.1bb63397.mjs';
import { o as overrideEnv } from '../shared/nuxi-edge.2b130854.mjs';
import { a as showVersions } from '../shared/nuxi-edge.2d608059.mjs';
import { d as defineNuxtCommand } from '../shared/nuxi-edge.8bffbe4c.mjs';
import '../shared/nuxi-edge.6fe47997.mjs';
import 'util';
import 'path';
import 'fs';
import 'os';
import 'tty';
import 'node:fs';
import '../shared/nuxi-edge.9bb12cfa.mjs';
import '../shared/nuxi-edge.4d9fcef2.mjs';
import 'node:module';
import 'node:url';
import 'assert';
import '../shared/nuxi-edge.d4f35867.mjs';

const buildCommand = defineNuxtCommand({
  meta: {
    name: "build",
    usage: "npx nuxi build [--prerender] [rootDir]",
    description: "Build nuxt for production deployment"
  },
  async invoke(args) {
    overrideEnv("production");
    const rootDir = resolve(args._[0] || ".");
    showVersions(rootDir);
    const { loadNuxt, buildNuxt, useNitro } = await loadKit(rootDir);
    const nuxt = await loadNuxt({
      rootDir,
      overrides: {
        _generate: args.prerender
      }
    });
    const nitro = useNitro?.();
    await clearDir(nuxt.options.buildDir);
    await writeTypes(nuxt);
    nuxt.hook("build:error", (err) => {
      consola.error("Nuxt Build Error:", err);
      process.exit(1);
    });
    await buildNuxt(nuxt);
    if (args.prerender) {
      const dir = nitro?.options.output.publicDir;
      const publicDir = dir ? relative(process.cwd(), dir) : ".output/public";
      consola.success(`You can now deploy \`${publicDir}\` to any static hosting!`);
    }
  }
});

export { buildCommand as default };
